src/components/RunGraph.vue

```diff
@@ -1,23 +1,49 @@
 <template>
-  <div ref="stage" class="run-graph" />
+  <div class="run-graph" :class="classes.root">
+    <div ref="stage" class="run-graph__stage" />
+    <div class="run-graph__actions">
+      <p-button title="Recenter Timeline" icon="Target" flat @click="() => centerViewport({ animate: true })" />
+      <p-button title="View Timeline in Fullscreen" icon="ArrowsPointingOutIcon" flat @click="toggleFullscreen" />
+    </div>
+  </div>
 </template>
 
 <script lang="ts" setup>
-  import { onBeforeUnmount, onMounted, ref } from 'vue'
+  import { computed, onBeforeUnmount, onMounted, ref } from 'vue'
   import { RunGraphProps } from '@/models/RunGraph'
-  import { ScaleXDomain, start, stop } from '@/objects'
+  import { ScaleXDomain, start, stop, centerViewport } from '@/objects'
   import { emitter } from '@/objects/events'
 
   // using the props object as a whole
   // eslint-disable-next-line vue/no-unused-properties
```

Comment:
this comment can probably be deleted now

Reply:
Nah eslint still complains about the props being unused. These are the reasons I don't like the compiler time macros. Their not real js/ts so they have to write all these special cases and handling into the ecosystem. And seems like the vue/no-unused-properties just isn't smart enough to tell I'm using the props const so all the props are used.